### PR TITLE
[opt](load) optimize the performance of row distribution

### PR DESCRIPTION
**Pull request by zclllyybb**
## Proposed changes

Issue Number: close #xxx

For non-pipeline non-sinkv2:
before: 14s
now: 6s-
For pipeline + sinkv2:
before: 230ms *48 instances
now: 38ms *48 instances

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by zclllyybb** [comment]
run buildall

**Comment by zclllyybb** [comment]
run buildall

**Comment by zclllyybb** [comment]
run buildall

File: be/src/vec/sink/writer/vtablet_writer.cpp [hunk 1]
```diff
@@ -1313,47 +1313,72 @@ Status VTabletWriter::_incremental_open_node_channel(
...
+
+        for (int index_num = 0; index_num < partition->indexes.size();
+             ++index_num) { // partition->indexes = [index, tablets...]
+
+            auto tablet_id = partition->indexes[index_num].tablets[tablet_index];
+            auto it = _channels[index_num]->_channels_by_tablet.find(
+                    tablet_id); // (tablet_id, VNodeChannel) where this tablet locate
+            std::vector<std::shared_ptr<VNodeChannel>>& tablet_locations = it->second;
+
+            DCHECK(it != _channels[index_num]->_channels_by_tablet.end())
```
Comment (HappenLee) [inline be/src/vec/sink/writer/vtablet_writer.cpp:1339]: DCHECK should before than `std::vector<std::shared_ptr<VNodeChannel>>& tablet_locations = it->second;`

**Comment by zclllyybb** [comment]
run buildall

**Comment by zclllyybb** [comment]
run buildall

**Comment by zclllyybb** [comment]
run buildall

**Comment by zclllyybb** [comment]
run buildall

**Comment by zclllyybb** [comment]
run buildall

**Comment by zclllyybb** [comment]
run buildall

**Comment by zclllyybb** [comment]
run buildall

**Review by dataroaring — DISMISSED** [review]
LGTM

**Review by BiteTheDDDDt — DISMISSED** [review]

File: be/src/vec/sink/writer/vtablet_writer.cpp [hunk 2]
```diff
@@ -1337,61 +1337,91 @@ Status VTabletWriter::_incremental_open_node_channel(
...
+            continue;
+        }
+        const auto& partition = partitions[row_idx];
+        const auto& tablet_index = tablet_indexes[row_idx];
+
+        for (int index_num = 0; index_num < partition->indexes.size();
+             ++index_num) { // partition->indexes = [index, tablets...]
+
+            auto tablet_id = partition->indexes[index_num].tablets[tablet_index];
+            LOG(WARNING) << "got tablet it " << tablet_id << " at row " << row_idx;
```
Comment (BiteTheDDDDt) [inline be/src/vec/sink/writer/vtablet_writer.cpp:1359]: maybe this log need removed?

Comment (zclllyybb) [reply be/src/vec/sink/writer/vtablet_writer.cpp:1359]: u r right

**Review by HappenLee — APPROVED** [review]
LGTM

**Review by BiteTheDDDDt — APPROVED** [review]

**Pull request merged**